### PR TITLE
Fix invalid Envoy fault bug

### PR DESCRIPTION
**Pull request by ijsnellf**
Fixes bug where special case was causing an invalid fault with no abort or delay to be generated. Resolves https://github.com/istio/pilot/issues/866.

### CODE REVIEW

File: proxy/envoy/fault.go [hunk 1]
```diff
@@ -40,14 +40,19 @@ func buildFaultFilters(routeConfig *HTTPRouteConfig) []HTTPFilter {
 
 // buildFaultFilter builds a single fault filter for envoy cluster
 func buildHTTPFaultFilter(cluster string, faultRule *proxyconfig.HTTPFaultInjection, headers Headers) *HTTPFilter {
+	abort, delay := buildAbortConfig(faultRule.Abort), buildDelayConfig(faultRule.Delay)
+	if abort == nil && delay == nil {
```
Comment (rkpagadala) [inline proxy/envoy/fault.go]: should this be || instead of &&

File: proxy/envoy/fault.go [hunk 2]
```diff
@@ -40,14 +40,19 @@ func buildFaultFilters(routeConfig *HTTPRouteConfig) []HTTPFilter {
 
 // buildFaultFilter builds a single fault filter for envoy cluster
 func buildHTTPFaultFilter(cluster string, faultRule *proxyconfig.HTTPFaultInjection, headers Headers) *HTTPFilter {
+	abort, delay := buildAbortConfig(faultRule.Abort), buildDelayConfig(faultRule.Delay)
```
Comment (rkpagadala) [inline proxy/envoy/fault.go]: Also the following reads better to me :)
abort := buildAbortConfig(faultRule.Abort)
delay := buildDelayConfig(faultRule.Delay)

File: proxy/envoy/fault.go (hunk 1 above)
Comment (ijsnellf) [inline proxy/envoy/fault.go]: No, it is valid if you have just 1 abort or 1 delay. The only invalid case is having neither.

File: proxy/envoy/fault.go (hunk 2 above)
Comment (ijsnellf) [inline proxy/envoy/fault.go]: Sure, bit easier to read.

**Pull request merged**